controllers/controllers.go

```diff
@@ -17,20 +22,34 @@ type Routes struct {
 	Public  *gin.RouterGroup
 }
 
+var tokenCache *cache.Cache
+
 func AuthenticationRequired(auths ...string) gin.HandlerFunc {
 	return func(c *gin.Context) {
-		// TODO
+		id_token := c.GetHeader("ID-Token")
```

Comment:
Minor nitpick, but we typically use camelCase rather than snake_case except for LOUD_SNAKE_CASE for constants